### PR TITLE
gh-95180: Add `TaskGroup` and `Runner` to AsyncIO API Index

### PR DESCRIPTION
**Pull request by siphc**
#95180

### CODE REVIEW

File: Misc/NEWS.d/next/Documentation/2022-07-24-05-48-14.gh-issue-95180.z8e649.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+Added :class:`TaskGroup` and :class:`Runner` to AsyncIO API Index
```
Comment (graingert) [inline Misc/NEWS.d/next/Documentation/2022-07-24-05-48-14.gh-issue-95180.z8e649.rst]:
```suggestion
Added :class:`asyncio.TaskGroup` and :class:`asyncio.Runner` to the asyncio High-level API Index
```

**Comment by kumaraditya303** [comment]
The change is very small no news entry is required, I added `skip news`/

File: Doc/library/asyncio-api-index.rst [hunk 2]
```diff
@@ -48,6 +48,12 @@ await on multiple things with timeouts.
     * - :class:`Task`
       - Task object.
 
+    * - :class:`TaskGroup`
+      - Holds a group of tasks.
+
+    * - :class:`Runner`
+      - Manages multiple function calls.
+
```
Comment (graingert) [inline Doc/library/asyncio-api-index.rst]: I'd recommend re-ordering these entries and rewording them eg:

```rst
Utilities to run asyncio programs, create Tasks, and
await on multiple things with timeouts.

.. list-table::
    :widths: 50 50
    :class: full-width-table

    * - :func:`run`
      - Create an event loop, run one async function, and close the loop.

    * - :class:`Runner`
      - A context manager to create an event loop, run one or more async functions, and close the loop.

    * - :class:`TaskGroup`
      - Combines a task creation API with a convenient and reliable way to wait for all tasks in the group to finish.

    * - :func:`create_task`
      - Start an asyncio Task.
```

**Review by graingert — CHANGES_REQUESTED** [review]

File: Doc/library/asyncio-api-index.rst (hunk 2 above)
Comment (graingert) [reply Doc/library/asyncio-api-index.rst]: specifically `Runner` should be introduced at the same time as `run` and `TaskGroup` should be introduced before `create_task`

**Comment by siphc** [comment]
Thanks! Let me know if it fits better now.

File: Doc/library/asyncio-api-index.rst [hunk 3]
```diff
@@ -19,10 +19,27 @@ await on multiple things with timeouts.
...
+      - Create an event loop, run one async function, and close the loop.
+
+    * - :class:`Runner`
+      - A context manager that simplifies multiple async function calls.
+
+    * - :class:`Task`
+      - Task object.
+
+    * - :class:`TaskGroup`
+      - A context manager that holds a group of tasks. Provides a way with
```
Comment (graingert) [inline Doc/library/asyncio-api-index.rst]:
```suggestion
      - A context manager that holds a group of tasks. Provides
```

File: Doc/library/asyncio-api-index.rst [hunk 4]
```diff
@@ -19,10 +19,27 @@ await on multiple things with timeouts.
     :class: full-width-table
 
     * - :func:`run`
-      - Create event loop, run a coroutine, close the loop.
+      - Create an event loop, run one async function, and close the loop.
```
Comment (kumaraditya303) [inline Doc/library/asyncio-api-index.rst]: It really runs a coroutine object not an async function. The previous wording was better.

Comment (gvanrossum) [reply Doc/library/asyncio-api-index.rst]: Agreed. The thing it runs is whatever the async function call returns, and such a thing is called a coroutine. We don't want people to think that they have to pass an async function name here.

**Review by gvanrossum — CHANGES_REQUESTED** [review]

**Review by gvanrossum — COMMENTED** [review]
In the issue, it was brought up that `timeout()` also belongs in this list. I agree.

**Comment by siphc** [comment]
I have made the requested changes; please review again.

**Pull request merged**